### DISCUSSION TITLE
Create Only Entity class for provided Table names in a Database

### DISCUSSION THREAD

**Discussion by ghost** [slick/slick#2638] [Questions]
Hi, I am kind of new in Slick and JDBC. I have to generate Entity Classes from given List of Tables of a DB.

I have been going through [Schema Code Generation](https://scala-slick.org/doc/3.4.1/code-generation.html) and some Slick basics. In the **customization** section of the page, there is some customization examples. From my understanding, I have to provide tables as the parameter here in the `H2Profile.createModel(...)`:
````
val db = Database.forConfig("h2mem1")
val modelAction = H2Profile.createModel(Some(H2Profile.defaultTables))
val modelFuture = db.run(modelAction)
````
Can you provide an example for this where I have to work with given list of table names or some kind of table references and make it of `Option[dbio.DBIO[Seq[MTable]]]` type before passing it as parameter here, not the `defaultTables`? Thanks

Also, I have been going through the [API](https://scala-slick.org/doc/3.4.1/codegen-api/slick/codegen/SourceCodeGenerator.html) for overrides. Specifically I was looking to disable other code generations and generate only entity classes. But so far, I didn't seem to find any. If it's possible, can you provide references or example? Thanks

Also, if all good, I wanted to know, say the entity class needs to extend some base class, how to do that? Thanks.

**Comment by nafg** [comment]
I would recommend giving my scalameta-based generator a try: https://github.com/nafg/slick-additions#slick-additions-codegen

My availability is limited in the immediate term but when I'm available I can help more.
